packages/x-charts/src/internals/plugins/featurePlugins/useChartCartesianAxis/useChartCartesianAxis.types.ts

```diff
@@ -47,6 +48,19 @@ export interface UseChartCartesianAxisParameters<S extends ScaleName = ScaleName
    * @param {null | ChartsAxisData} data The data about the clicked axis and items associated with it.
    */
   onAxisClick?: (event: MouseEvent, data: null | ChartsAxisData) => void;
+  /**
+   * The function called when the pointer position corresponds to a new axis data item.
+   * This update can either be caused by a pointer movement, or an axis update.
+   * In case of multiple axes, the function get called if at least ones axis is updated.
+   * And the parameter contains the identifier for all axes with a `data` property.
+   * @param {AxisItemIdentifier[]} newAxisItems The array of axes item identifiers.
+   */
+  onAxisInteraction?: (newAxisItems: AxisItemIdentifier[]) => void;
+  /**
+   * The controlled axis highlighted.
+   * Indicates the direction, axis id, and data index to highlight.
+   */
+  highlightedAxis?: AxisItemIdentifier | null;
```

Comment:
Isn't it a bit weird that the prop that sets the highlighted is called `highlightedAxis`, but when it changes, the event handler is called `onAxisInteraction`?

Should they be `axisInteraction`/`onAxisInteraction` or `highlightedAxis`/`onAxisHighlight`?

Reply:
> My concern is the upcoming tooltip control that will need the same callback to know which axis needs to be displayed.

Sorry, what do you mean? 

> But we could effectively expose an `onAxisHighlight` and trigger another function for the tooltip.

Ah, do you mean `onAxisHighlight` for axis highlights and `onTooltipChange` when the tooltip should display different items? 

> Should onAxisHighlight get an array of axis item or just the first one.

I think that's what I'm trying to get at in my comment below. I think there might be more than one axis highlighted at a time, so we probably need an array.